notebook/static/notebook/js/notebook.js

```diff
@@ -3037,6 +3037,37 @@ define(function (require) {
         this.events.trigger('checkpoint_deleted.Notebook');
         this.load_notebook(this.notebook_path);
     };
+    
+    /**
+    * Switch the keymap of all codemirror code cell into the state 
+    * specified by the argument. 
+    *
+    * @param {string} keymap in {'vim', 'sublime', 'default', 'emacs'}
+    */
+    Notebook.prototype.switch_keymap_to = function(keymap) {
+        var vimMode = true;
+        var cells = this.get_cells();
+        var cell = cells[1];
+        if (keymap === "vim"){
+            vimMode = true;
+            this.keyboard_manager.mode = 'vim-command';
+        } else if ( keymap === 'sublime' || keymap === 'default') {
+            vimMode = false;
```

Comment:
if you set vimMode to `null`, the update option will remove the key from the config file, which is cleaner I think.